### PR TITLE
obs2ioda cris and iasi update

### LINKED ISSUE #1364: add hyperspectral additions for IASI and CrIS to ncar bufr2ioda
## Description

a few updates are needed for the bufr2ioda converter for the IASI and CrIS sensors
including NOAA-21 and some observation error estimates

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

some updates to the modules in ncar obs2ioda for IASI and CrIS

## Issue(s) addressed

Resolves #1364 

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by BenjaminRuston** [comment]
@ibanos90 we definitely should find some time to sync up on this, don't want to diverge and hopefully can consolidate at some point, but including you to show you what we've done for the IASI and CrIS

File: src/ncar-bufr2nc-fortran/radiance_mod.f90 [hunk 1]
```diff
@@ -940,11 +949,12 @@ subroutine read_cris(filename, filedate)
...
 
-            chan_loop: do i = 1, nchan
-               if (data1b8(1, i) > r8bfms .or. data1b8(2, i) > r8bfms) cycle chan_loop
-               rlink%ch(i) = nint(data1b8(1, i))
-               rlink%tb(i) = data1b8(2, i)*1000.0  ! radiance for now
-            end do chan_loop
+            if ( nchan > 0 ) then
+               chan_loop: do i = 1, nchan
+                  if ( data1b8(1,i) < r8bfms ) rlink % ch(i) = nint(data1b8(1,i))
+                  if ( data1b8(2,i) < r8bfms ) rlink % tb(i) = data1b8(2,i) * 1000.0  ! radiance for now
```
Comment (fabiolrdiniz) [inline src/ncar-bufr2nc-fortran/radiance_mod.f90:955]: I wonder if we shouldn't play safe...
```suggestion
                  if ( data1b8(1,i) /= r8bfms ) rlink % ch(i) = nint(data1b8(1,i))
                  if ( data1b8(2,i) /= r8bfms ) rlink % tb(i) = data1b8(2,i) * 1000.0  ! radiance for now
```

**Review by PatNichols — APPROVED** [review]
The syntax errors are fixed and I approve.

**Review by ashley314 — APPROVED** [review]
I tested with IASI and the files were able to be converted. Thanks Ben!

**Comment by PatNichols** [comment]
@ashley314 @BenjaminRuston Can we take this out of draft status? It seems ready to go??

**Pull request merged**